perfectsense/brightspot-cms
# RTC disconnect detection: BSP-2383, BSP-2382

Test Scenarios:

Offline Test:

1. Log in with account A in Chrome
1. Log in with account B in Firefox
1. Open a piece of content in Chrome
1. Open the same piece of content in Firefox
1. Change a field in Chrome
1. Verify that that field is locked in Firefox
1. Open the web inspector in Chrome
1. Go to the Network tab
1. Click on "No throttling" drop-down and select "Offline" preset
1. Wait about 30 seconds
1. Verify that both the viewer and the field lock goes away in Firefox
1. Restore "No throttling"
1. Verify that both the viewer and the field lock reappears in Firefox

Connection Sharing Test:

1. Log in with account A in Chrome
1. Log in with account B in Firefox
1. Open a piece of content in Chrome
1. Open the same piece of content in Firefox
1. Verify that field locking works
1. Open another piece of content in Chrome in another tab
1. Open the same piece of content in Firefox in another tab
1. Verify that field locking works
1. Open web inspector in both tabs of Chrome
1. Go to the Network tab and filter by "_rtc"
1. Verify that only the first tab contains any entries
1. Close the first tab
1. Verify that now the second tab contains entries
1. Verify that the field locking works on the second content
